src/main/java/xyz/nucleoid/plasmid/game/common/team/TeamManager.java

```diff
@@ -0,0 +1,330 @@
+            this.teamToPlayerSet.get(team).remove(ref);
+
+            Team vanilla = this.getVanilla(team);
+            vanilla.getPlayerList().remove(ref.name());
+            if (this.gameActivity != null) {
+                Packet<?> packet = TeamS2CPacket.changePlayerTeam(vanilla, ref.name(), TeamS2CPacket.Operation.REMOVE);
+                this.gameActivity.getGameSpace().getPlayers().sendPacket(packet);
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Removes player from team
+     *
+     * @param player
+     * @return true if player was removed successfully
+     */
+    public boolean removePlayerTeam(ServerPlayerEntity player) {
+        return this.removePlayerTeam(PlayerRef.of(player));
+    }
+
+    /**
+     * Returns team player is in
+     *
+     * @param player Player reference
+     * @return GameTeam or null
+     */
+    @Nullable
+    public GameTeam getTeamOf(PlayerRef player) {
+        return this.playerRefToTeam.get(player);
+    }
+
+    /**
+     * Returns team player is in
+     *
+     * @param player
+     * @return GameTeam or null
+     */
+    @Nullable
+    public GameTeam getTeamOf(ServerPlayerEntity player) {
+        return this.playerRefToTeam.get(PlayerRef.of(player));
+    }
+
+    /**
+     * Gets set of players being part of the team
+     *
+     * @param team targeted {@link GameTeam}
+     * @return Set of Players
+     */
+    public PlayerSet getPlayers(GameTeam team) {
+        PlayerSet set = this.teamToPlayerSet.get(team);
+
+        return set == null ? PlayerSet.EMPTY : set;
+    }
+
+    /**
+     * Gets set of {@link PlayerRef} being part of the team
+     *
+     * @param team targeted {@link GameTeam}
+     * @return Set of {@link PlayerRef}
+     */
+    public Set<PlayerRef> getPlayerRefs(GameTeam team) {
+        Set<PlayerRef> set = this.teamToPlayerRefSet.get(team);
+
+        return set == null ? Collections.emptySet() : set;
+    }
+
+    public boolean getFriendlyFire(GameTeam team) {
+        return this.teamFriendlyFire.getBoolean(team);
+    }
+
+    public boolean setFriendlyFire(GameTeam team, boolean value) {
+        return this.teamFriendlyFire.put(team, value);
+    }
+
+    public AbstractTeam.CollisionRule getCollisionRule(GameTeam team) {
+        return this.getVanilla(team).getCollisionRule();
+    }
+
+    public void setCollisionRule(GameTeam team, AbstractTeam.CollisionRule rule) {
+        this.getVanilla(team).setCollisionRule(rule);
+        this.sendUpdates(team);
+    }
+
+    public AbstractTeam.VisibilityRule getNameTagVisibilityRule(GameTeam team) {
+        return this.getVanilla(team).getNameTagVisibilityRule();
+    }
+
+    public void setNameTagVisibilityRule(GameTeam team, AbstractTeam.VisibilityRule rule) {
+        this.getVanilla(team).setNameTagVisibilityRule(rule);
+        this.sendUpdates(team);
+    }
+
+    public Text applyPlayerFormatting(PlayerRef player, Text text) {
```

Comment:
It isn't entirely clear what formatting this is or what it's being applied to- is this the player name?

Reply:
Yep, it's player name and formatting is samish to one from vanilla teams

Reply:
Maybe worth renaming to like `formatPlayerName` or similar?